src/hera/expr/_node.py

```diff
@@ -0,0 +1,224 @@
+from typing import Any, Dict, List, Optional, Union
+
+Primitives = Optional[Union[str, int, bool, float, range]]
+Arrays = List[Any]
+Maps = Dict[Primitives, Any]
+Constants = Union[Primitives, Arrays, Maps]
+
+# Mapping what is available at
+# https://github.com/antonmedv/expr/blob/34d2cbe7e43af1348d7010d951f1d7fe44a91500/checker/checker.go#L198
+BINARY_OP_MAP = {
+    "__add__": "+",
+    "__and__": "&&",
+    "__eq__": "==",
+    "__ge__": ">=",
+    "__gt__": ">",
+    "__le__": "<=",
+    "__lt__": "<",
+    "__mod__": "%",
+    "__mul__": "*",
+    "__ne__": "!=",
+    "__or__": "||",
+    "__pow__": "**",
+    "__sub__": "-",
+    "__truediv__": "/",
+    "contains": "contains",
+    "ends_with": "endsWith",
+    "in_": "in",
+    "matches": "matches",
+    "not_in": "not in",
+    "starts_with": "startsWith",
+}
+
+BUILTINS = (
+    "all",
+    "any",
+    "count",
+    "filter",
+    "map",
+    "none",
+    "one",
+)
+
+
+class Node:
+    def __getattr__(self, name: str) -> "GetAttr":
+        return GetAttr(self, name)
+
+    def __getitem__(self, name: str) -> "GetItem":
+        return GetItem(self, name)
+
+    def __invert__(self):
+        return UnaryOp(self, "!")
+
+    def __neg__(self):
+        return UnaryOp(self, "-")
+
+    def __pos__(self):
+        return UnaryOp(self, "+")
+
+    def length(self):
+        return Callable("len", self)
+
+    def as_float(self):
+        return Callable("asFloat", self)
+
+    def as_int(self):
+        return Callable("asInt", self)
+
+    def check(self, truthy_value: 'Node', falsey_value: 'Node') -> 'Check':
```

Comment:
Nit - I think this could be `falsy` by majority rule (it is, still, grammatically correct)
```suggestion
    def check(self, truthy_value: 'Node', falsy_value: 'Node') -> 'Check':
```